### ISSUE TITLE
Preview Disabled Product & Multiples languages 

### ISSUE THREAD

**Issue by Progi1984** [PrestaShop/PrestaShop#34191]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

When I duplicate a product in multilingual shop and I preview it in FrontOffice, I can't change the language

### Expected behavior

When I duplicate a product in multilingual shop and I preview it in FrontOffice, I can change the language

### Steps to reproduce

1. Install a shop in English / France
2. Go to BO > Catalog > Products
3. Click on Preview for the first line (for example) : Customizable mug
4. :heavy_check_mark: That go to FrontOffice > Product Page in English
5. Change the language to French 
6.  :heavy_check_mark: That go to FrontOffice > Product Page in French
7. Return to BO
8. Duplicate this product
9. :heavy_check_mark: It's disabled
10. Click on Preview for the first line (for example) : Customizable mug
11.  :heavy_check_mark: That go to FrontOffice > Product Page in English
12. Change the language to French
13. :red_circle: That go to FrontOffice > Categories

### PrestaShop version(s) where the bug happened

9.0.x

### PHP version(s) where the bug happened

8.1

### Your company or customer's name goes here (if applicable).

@PrestaShopCorp

**Comment by Hlavtox** [comment]
It doesn't have to do anything with duplication. :-)

1. Have two languages.
2. Disable any product in BO. Preview it.
3. Change language - 404/redirect, whatever, because `adtoken=1f8fefacf909891d2342f34ed38ad920&id_employee=1&preview=1 `is not in the link.

**Comment by AureRita** [comment]
Hi @Progi1984 

Thank you for your report, I tried to reproduce your issue but currently I can't as you can see : 

[Untitled_ Oct 6, 2023 4_53 PM.webm](https://github.com/PrestaShop/PrestaShop/assets/100699445/0a5a0082-6d16-4ee3-a9fc-f6e04685631c)

Did I miss something ? 

Waiting for your feedback

**Comment by Progi1984** [comment]
@AureRita Yes you change the language on the BackOffice, not on the frontoffice.

**Comment by AureRita** [comment]
@Progi1984 

Thank you for the detail, I reproduce the issue with PrestaShop version 9.0.x, as we can see here :

[Untitled_ Oct 6, 2023 5_20 PM.webm](https://github.com/PrestaShop/PrestaShop/assets/100699445/c7d5f522-21c4-4833-b747-58526084e672)

I'll add this to the backlog so it can be fixed.

Please be aware that some issues might take a very long time to be resolved

Thanks
